### PR TITLE
feat(expression): numeric/string can  cast to boolean when it's in filter

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

1. feat(expression): numeric/string can  cast to boolean when it's in filter
2. fix nullable with div by zero corner case 
3. improve bitmap collect from trusted iterator

Fixes #issue

### CODE REVIEW

**Comment by sundy-li** [comment]
Looks ambiguous after this auto-cast feature.
```
select  not 33;  -- false
select not 'abc'; -- false
select 3 and 'abc'; -- false
select false or 'ab'; -- true
```

I would like to keep the old way.

**Comment by sundy-li** [comment]
enable it in the filter kernel.

File: src/query/expression/src/kernels/filter.rs [hunk 1]
```diff
@@ -84,21 +87,42 @@ impl Chunk {
     // Must be nullable boolean or boolean value
```
Comment (andylokandy) [inline src/query/expression/src/kernels/filter.rs:87]: should update the comment

File: src/query/expression/src/kernels/filter.rs [hunk 2]
```diff
@@ -84,7 +84,7 @@ impl Chunk {
         }
     }
 
-    // Must be nullable boolean or boolean value
+    // Must be numeric/boolean or string value type
```
Comment (andylokandy) [inline src/query/expression/src/kernels/filter.rs:87]:
```suggestion
    // Must be numeric, boolean, or string value type
```

**Review by andylokandy — APPROVED** [review]

**Comment by sundy-li** [comment]
@mergify update

**Pull request merged**